frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,309 @@
+import * as React from 'react';
+import {
+  Button,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStatePrimary,
+  Menu,
+  MenuContent,
+  MenuGroup,
+  MenuInput,
+  MenuItem,
+  MenuList,
+  Switch,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useUserSettingsCompatibility } from '@console/shared';
+import { isSystemNamespace } from './factory/table-filters';
+import NamespaceMenuToggle from './namespace-menu-toggle';
+
+const NamespaceBarDropdown_ = (props: NamespaceBarDropdownProps) => {
+  const { t } = useTranslation();
+  const {
+    actions,
+    disabled,
+    isProjects,
+    onSelect,
+    options,
+    selected,
+    shortCut,
+    storageKey,
+    title,
+    userSettingsPrefix,
+  } = props;
+
+  const MAX_HEIGHT = '60vh';
+
+  const filterRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+
+  const [isMainSelectOpen, setMainSelectOpen] = React.useState(false);
+  const [filterText, setFilterText] = React.useState('');
+  const [activeOption, setActiveOption] = React.useState(selected);
+  const [footerHeight, setFooterHeight] = React.useState(0);
+
+  React.useEffect(() => {
+    setActiveOption(selected);
+  }, [selected, options]);
+
+  // Bookmarking / favorites (note in <= 4.8 this feature was known as bookmarking)
+  const bookmarkUserSettingsKey = userSettingsPrefix
+    ? `${userSettingsPrefix}.bookmarks`
+    : undefined;
+
+  const systemNamespacesInSelectSettingsKey = userSettingsPrefix
+    ? `${userSettingsPrefix}.systemNamespaceInSelect`
+    : undefined;
+
+  const bookmarkStorageKey = storageKey ? `${storageKey}-bookmarks` : undefined;
+  const systemNamespaceInSelectKey = storageKey
+    ? `${storageKey}-systemNamespaceInSelect`
+    : undefined;
+
+  const [bookmarks, setBookmarks] = useUserSettingsCompatibility(
+    bookmarkUserSettingsKey,
+    bookmarkStorageKey,
+    undefined,
+    true,
+  );
+
+  const [systemNamespacesInSelect, setSystemNamespacesInSelect] = useUserSettingsCompatibility(
+    systemNamespacesInSelectSettingsKey,
+    systemNamespaceInSelectKey,
+    false,
+    true,
+  );
+
+  const onBookmark = React.useCallback(
+    (key, active) => {
+      setBookmarks((oldBookmarks) => ({ ...oldBookmarks, [key]: active ? true : undefined }));
+    },
+    [setBookmarks],
+  );
+
+  const areSystemNamespaces = () => {
+    return options.some((option) => isSystemNamespace(option));
+  };
+
+  React.useEffect(() => {
+    if (isMainSelectOpen === true) {
+      setFilterText('');
+    }
+  }, [isMainSelectOpen]);
+
+  const onFavorite = (event: any, itemId: string) => {
+    const isCurrentBookmark = bookmarks && bookmarks[itemId] === true;
```

Comment:
Do we need to check specifically for `true` or is any truthy value OK? This could just be

```suggestion
    const isCurrentBookmark = bookmarks?.[itemId];
```

Reply:
Looks good.  Change on next commit.